hotel/models/hotel_folio.py

```diff
@@ -0,0 +1,810 @@
+        This method gives the duration between check in and checkout
+        if customer will leave only for some hour it would be considers
+        as a whole day.If customer will check in checkout for more or equal
+        hours, which configured in company as additional hours than it would
+        be consider as full days
+        --------------------------------------------------------------------
+        @param self: object pointer
+        @return: Duration and checkout_date
+        """
+        configured_addition_hours = 0
+        wid = self.warehouse_id
+        whouse_com_id = wid or wid.company_id
+        if whouse_com_id:
+            configured_addition_hours = wid.company_id.additional_hours
+        myduration = 0
+        if self.checkout_date and self.checkin_date:
+            dur = self.checkin_date - self.checkin_date
+            sec_dur = dur.seconds
+            if (not dur.days and not sec_dur) or (dur.days and not sec_dur):
+                myduration = dur.days
+            else:
+                myduration = dur.days + 1
+            # To calculate additional hours in hotel room as per minutes
+            if configured_addition_hours > 0:
+                additional_hours = abs((dur.seconds / 60) / 60)
+                if additional_hours >= configured_addition_hours:
+                    myduration += 1
+        self.duration = myduration
+        self.duration_dummy = self.duration
+
+    @api.model
+    def create(self, vals):
+        """
+        Overrides orm create method.
+        @param self: The object pointer
+        @param vals: dictionary of fields value.
+        @return: new record set for hotel folio.
+        """
+        if not "service_lines" and "folio_id" in vals:
+            tmp_room_lines = vals.get("room_lines", [])
+            vals["order_policy"] = vals.get("hotel_policy", "manual")
+            vals.update({"room_lines": []})
+            folio_id = super(HotelFolio, self).create(vals)
+            for line in tmp_room_lines:
+                line[2].update({"folio_id": folio_id.id})
+            vals.update({"room_lines": tmp_room_lines})
+            folio_id.write(vals)
+        else:
+            if not vals:
+                vals = {}
+            vals["name"] = self.env["ir.sequence"].next_by_code("hotel.folio")
+            vals["duration"] = vals.get("duration", 0.0) or vals.get(
+                "duration_dummy", 0.0
+            )
+            folio_id = super(HotelFolio, self).create(vals)
+            folio_room_line_obj = self.env["folio.room.line"]
+            h_room_obj = self.env["hotel.room"]
+            try:
+                for rec in folio_id:
+                    if not rec.reservation_id:
+                        for room_rec in rec.room_lines:
+                            prod = room_rec.product_id.name
+                            room_obj = h_room_obj.search([("name", "=", prod)])
+                            room_obj.write({"isroom": False})
```

Comment:
room = h_room_obj.search([("product_id", "=", room_rec.product_id.id)])
room.write({"isroom": False})

Get a more accurate result if we searching with ids instead of the name

Reply:
Done.